### PR TITLE
chore: Make set manifest state more understandable

### PR DESCRIPTION
**Pull request by ruanxin**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: internal/declarative/v2/reconciler.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ var (
 	ErrManagerInErrorState            = errors.New("manager is in error state")
 	ErrResourceSyncDiffInSameOCILayer = errors.New("resource syncTarget diff detected but in " +
 		"same oci layer, prevent sync resource to be deleted")
+	ErrStateRequireUpdate = errors.New("manifest state requires update")
```
Comment (lindnerby) [inline internal/declarative/v2/reconciler.go:37]: pkg private when not used outside

File: internal/manifest/status/condition.go [hunk 2]
```diff
@@ -74,3 +74,14 @@ func ConfirmResourcesCondition(manifest *v1beta2.Manifest) {
 		manifest.SetStatus(status.WithOperation(resourceCondition.Message))
 	}
 }
+
+func ConfirmInstallationCondition(manifest *v1beta2.Manifest) {
```
Comment (lindnerby) [inline internal/manifest/status/condition.go:78]: quick and easy unit tests can be written for this func

File: internal/manifest/status/state.go [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
-		installationCondition.Status = apimetav1.ConditionTrue
-		meta.SetStatusCondition(&status.Conditions, installationCondition)
+const (
+	ResourcesAreReadyMsg   = "resources are ready"
+	waitingForResourcesMsg = "waiting for resources to become ready"
+)
 
-		manifest.SetStatus(status.WithState(newState).WithOperation(installationCondition.Message))
-		return ErrInstallationConditionRequiresUpdate
+func RequireManifestStateUpdateAfterSyncResource(manifest *v1beta2.Manifest, newState shared.State) bool {
```
Comment (lindnerby) [inline internal/manifest/status/state.go:13]: unit tests for this func would be nice

**Review by lindnerby — CHANGES_REQUESTED** [review]
lgtm. just some minor comments

File: internal/manifest/status/state.go (hunk 3 above)
Comment (ruanxin) [reply internal/manifest/status/state.go:13]: done

File: internal/manifest/status/condition.go (hunk 2 above)
Comment (ruanxin) [reply internal/manifest/status/condition.go:78]: done

File: internal/declarative/v2/reconciler.go (hunk 1 above)
Comment (ruanxin) [reply internal/declarative/v2/reconciler.go:37]: done

**Review by lindnerby — APPROVED** [review]

**Pull request merged**